src/Groups/abelian_aut.jl

```diff
@@ -105,19 +111,19 @@ function _as_subgroup(aut::AutomorphismGroup{T}, subgrp::GapObj, ::Type{S}) wher
 end
 
 """
-    hom(f::AutomorphismGroupElem{T}) where T
+    hom(f::AutomorphismGroupElem{GrpAbFinGen}) -> GrpAbFinGenMap 
 
 Return the element `f` of type `GrpAbFinGenMap`.
 """
-function hom(f::AutomorphismGroupElem{T}) where T <: GrpAbFinGen
-  A = parent(f).G
+function hom(f::AutomorphismGroupElem{GrpAbFinGen}) 
+  A = group(parent(f))
   imgs = [f(a) for a in gens(A)]
   return hom(A, A, imgs)
 end
 
 
-function (aut::AutomorphismGroup{T})(f::GrpAbFinGenMap) where T
-  @assert isautomorphism(aut.G,f.map) "Map doesn't define an element of AutomorphismGroup"
+function (aut::AutomorphismGroup{GrpAbFinGen})(f::GrpAbFinGenMap)
+  defines_automorphism(group(aut),matrix(f)) || error("Map does not define an automorphism of the abelian group.")
```

Comment:
`isbijective(f)`

Reply:
This is what I meant to do at the beginning, here we need to ensure that the domain of `f` and the automorphism domain of `aut` match.

Reply:
You are not checking that. You are just taking the "matrix" and test if it defines and automorphism for `group(aut)`. There is no interaction with `domain(f)` or `codomain(f)`.

The check should be `domain(f) === codomain(f) && isbijective(f)`.

Reply:
And is it possible to "triple" the check ? like `domain(f) === codomain(f) && domain(f) === group(aut) && isbijective(f)` ? 
I thought that the method was doing roughly the same since by construction `f = hom(domain(f), codomain(f), f.map)`, but maybe this task takes more time.

Reply:
Yes, you can just do `(domain(f) === codomain(f) && domain(f) === group(aut) && isbijective(f)) || error("asd")`